tests/tests.go

```diff
@@ -14,6 +14,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
```

Comment:
I don't understand why the original code is not sufficient. 

I mean, 
* the old leader was deleted for sure (otherwise we would have failed the Eventually [here](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/151/commits/e7e4217b541adcc0db556ef99be20cd0c3275e14#diff-4de094b4f2274c29e9f11102da4ca7d0L231))
* The pod is checked [here](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/151/commits/e7e4217b541adcc0db556ef99be20cd0c3275e14#diff-4de094b4f2274c29e9f11102da4ca7d0L245) for readiness (second part of if condition)
* the're no way that the non-leader pod got ready.
 
can you specify why exactly it did not work?

Reply:
Leader pod can exists and have the leader label but not be ready yet, so we continue with a non functional leader.

Reply:
but isn't the [second condition](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/151/files#diff-4de094b4f2274c29e9f11102da4ca7d0L263) enough to check readiness?

Reply:
issue is there is a time when we have three kmp pods: two running and one terminating
and Terminating is at Ready since webhook is already running there

So there is a point to this PR.